app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -578,6 +593,45 @@ public void afterTextChanged(Editable s) {
         }
 
         textEditor.requestFocus();
+
+        if (inReplyToId == null) {
+            composeAvatar.setOnClickListener(v -> showAccountSwitchDialog(true));
+
+            if (!intent.hasExtra(ACCOUNT_PRESET) && !intent.hasExtra(SAVED_TOOT_UID_EXTRA) && (getCallingActivity() == null || !getCallingActivity().getClassName().equals(MainActivity.class.getName()))) {
+                showAccountSwitchDialog(false);
+            }
+        }
+    }
+
+    private void showAccountSwitchDialog(boolean cancelable) {
+        LayoutInflater inflater = (LayoutInflater)getSystemService(LAYOUT_INFLATER_SERVICE);
+        LinearLayout dialogView = (LinearLayout) inflater.inflate(R.layout.account_selection_dialog, null, false);
+        RecyclerView recyclerView = dialogView.findViewById(R.id.account_list);
+
+        recyclerView.setHasFixedSize(true);
+        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
+        recyclerView.setLayoutManager(layoutManager);
+        AlertDialog.Builder builder = new AlertDialog.Builder(this)
+                .setView(dialogView);
+
+        if (!cancelable) {
+            builder.setOnCancelListener(dialogInterface -> finish());
+        }
+
+        AlertDialog dialog = builder.create();
+        AccountListAdapter adapter = new AccountListAdapter(accountManager.getAllAccountsOrderedByActive(), account -> {
+            accountManager.setActiveAccount(account);
```

Comment:
Maybe I'm missing the point, but why do we have to restart activity? What happens after you return from it - all other screens are still for the old account (or flags handle this)?
Maybe we can avoid doing `setActiveAccount()` by using `InstanceSwitchAuthInterceptor`? Need to fix search too in this case...
I would ask input of others on how we should do it.
Are you in the Matrix room?